tests/src/main/java/org/hibernate/beanvalidation/tck/tests/methodvalidation/ValidateParametersTest.java

```diff
@@ -236,22 +226,17 @@ public void testCrossParameterConstraintGivenSeveralTimes() throws Exception {
 				parameterValues
 		);
 
-		assertCorrectNumberOfViolations( violations, 2 );
+		assertNumberOfViolations( violations, 2 );
 
 		assertCorrectConstraintTypes(
 				violations,
 				MyCrossParameterConstraint.class,
 				MyCrossParameterConstraint.class
 		);
-		assertCorrectPathNodeNames(
-				violations,
-				names( methodName, TestUtil.CROSS_PARAMETER_NODE_NAME ),
-				names( methodName, TestUtil.CROSS_PARAMETER_NODE_NAME )
-		);
-		assertCorrectPathNodeKinds(
-				violations,
-				kinds( ElementKind.METHOD, ElementKind.CROSS_PARAMETER ),
-				kinds( ElementKind.METHOD, ElementKind.CROSS_PARAMETER )
+		assertThat( violations ).containsOnlyPaths(
```

Comment:
This is why we need to check the number of violations. containsOnly deduplicates the elements.